### PR TITLE
Fix gtest invoking while workdir is set

### PR DESCRIPTION
**Pull request by skartashev**
Fixes: https://github.com/mesonbuild/meson/issues/7903

### CODE REVIEW

File: mesonbuild/mtest.py [hunk 1]
```diff
@@ -697,10 +697,10 @@ def preexec_fn() -> None:
 
         extra_cmd = []  # type: T.List[str]
         if self.test.protocol is TestProtocol.GTEST:
-            gtestname = '{}.xml'.format(self.test.name)
+            gtestname = '{}'.format(self.test.name)
             if self.test.workdir:
-                gtestname = '{}:{}'.format(self.test.workdir, self.test.name)
-            extra_cmd.append('--gtest_output=xml:{}'.format(gtestname))
+                gtestname = '{}/{}'.format(self.test.workdir, self.test.name)
```
Comment (dcbaker) [inline mesonbuild/mtest.py:702]: This should use `os.path.join`, instead of hard coding `/`.

File: mesonbuild/mtest.py [hunk 2]
```diff
@@ -697,10 +697,10 @@ def preexec_fn() -> None:
 
         extra_cmd = []  # type: T.List[str]
         if self.test.protocol is TestProtocol.GTEST:
-            gtestname = '{}.xml'.format(self.test.name)
+            gtestname = '{}'.format(self.test.name)
```
Comment (dcbaker) [inline mesonbuild/mtest.py:700]: Either leave the .xml here, or replace this with `gtestname = self.test.name`

**Review by dcbaker — CHANGES_REQUESTED** [review]
This looks like legitimate bug fix, so we should try to get this into 0.56.0, I've got a few nits I'd like changed first.

File: mesonbuild/mtest.py (hunk 1 above)
Comment (skartashev) [reply mesonbuild/mtest.py]: Fixed

File: mesonbuild/mtest.py (hunk 2 above)
Comment (skartashev) [reply mesonbuild/mtest.py]: Fixed

**Comment by skartashev** [comment]
[gtest_failure.zip](https://github.com/mesonbuild/meson/files/5453719/gtest_failure.zip)

Added test with/without workdir

**Review by dcbaker — APPROVED** [review]

**Comment by dcbaker** [comment]
looks good, thanks! I'll merge when CI comes back green (I don't see why it wouldn't).

**Comment by dcbaker** [comment]
the swift failures in the 16.04 image are not related to this MR.

**Pull request merged**